File: ai-service/app/schemas/diagnosis.py
from pydantic import BaseModel


class DiagnosisResponse(BaseModel):
    plant: str
    disease: str
    confidence: float
    status: str
    explanation: str
    treatment: list[str]


def compute_status(confidence: float) -> str:
    if confidence > 0.90:
        return "HIGH_CONFIDENCE"
    if confidence >= 0.70:
        return "MEDIUM_CONFIDENCE"
    return "LOW_CONFIDENCE"

File: mobile/src/services/diagnosis.ts
import { apiClient } from "../api/client";

export interface DiagnosisResult {
  diagnosisId: string;
  plant: string;
  disease: string;
  confidence: number;
  status: string;
  explanation: string;
  treatment: string[];
  createdAt: string;
}

export async function getDiagnosis(imageUrl: string): Promise<DiagnosisResult> {
  console.log(`[Diagnosis] Requesting diagnosis for: ${imageUrl}`);
  const res = await apiClient.post("/diagnosis", { imageUrl });
  console.log(`[Diagnosis] Received — plant: ${res.data.plant}, disease: ${res.data.disease}, id: ${res.data.diagnosisId}`);
  return res.data;
}
